app/code/Magento/Ui/Config/Data.php

```diff
@@ -64,26 +65,37 @@ class Data implements \Magento\Framework\Config\DataInterface
      */
     private $argumentInterpreter;
 
+    /**
+     * Store manager.
+     *
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
     /**
      * @param string $componentName
      * @param ReaderFactory $readerFactory
      * @param CacheInterface $cache
      * @param SerializerInterface $serializer
      * @param InterpreterInterface $argumentInterpreter,
+     * @param StoreManagerInterface|null $storeManager
      */
     public function __construct(
         $componentName,
         ReaderFactory $readerFactory,
         CacheInterface $cache,
         SerializerInterface $serializer,
-        InterpreterInterface $argumentInterpreter
+        InterpreterInterface $argumentInterpreter,
+        StoreManagerInterface $storeManager = null
     ) {
         $this->readerFactory = $readerFactory;
         $this->cache = $cache;
         $this->serializer = $serializer;
         $this->componentName = $componentName;
         $this->argumentInterpreter = $argumentInterpreter;
-        $this->cacheId = static::CACHE_ID . '_' . $componentName;
+        $this->storeManager = $storeManager ?:
+            \Magento\Framework\App\ObjectManager::getInstance()->get(StoreManagerInterface::class);
+        $this->cacheId = static::CACHE_ID . '_' . $componentName . '_' . $this->storeManager->getStore()->getCode();
```

Comment:
Please use store_id instead, with code there could be a collision and also it is usually longer.

Reply:
Done